### PR TITLE
Adding create meeting command to MS teams

### PR DESCRIPTION
**Pull request by ohaim1008**
Docker image for teams can't be raised - demisto/etc#33818

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/31658

## Description
Adding creating meeting command.
Adding GrantAccessPolicy script, this script should help a customer to set up the user's needed permissions in order to create a meeting.

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py
Comment (Itay4) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:879]: `/me` works only when using `delegated` permissions, we use app permissions

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.yml
Comment (Itay4) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.yml:334]:
```suggestion
      description: The meeting participants
```

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.yml
Comment (Itay4) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.yml:337]:
```suggestion
      description: The display name of the participants
```

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.yml
Comment (Itay4) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.yml:339]: did you verify the integration works as expected with this docker image?
see https://github.com/demisto/etc/issues/33818 for more details

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.yml
Comment (Itay4) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.yml:296]: please elaborate on the date format and add example

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.yml
Comment (Itay4) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.yml:302]: please elaborate on the date format and add example
also, why is this mandatory ?

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py
Comment (Itay4) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:909]: please use the walrus operator here

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py
Comment (Itay4) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:932]: please add return type hint

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py
Comment (Itay4) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:938]: please be consistent with `"` and `'`

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py
Comment (Itay4) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:941]: please use walrus operator

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py
Comment (Itay4) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:948]: better to just use `.get()` instead of 4 conditions

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py
Comment (Itay4) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:916]: please be consistent with `"` and `'`

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py
Comment (Itay4) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:924]: why are you returning `CommandResult` ? `return_results` does not call this func

**Comment by Itay4** [comment]
please add unit test

**Comment by Itay4** [comment]
this requires additional permissions, please mention it in the release notes and update the readme accordingly

**Comment by ohaim1008** [comment]
The docker image in the script `GrantAccessPolicy` will change after https://github.com/demisto/dockerfiles/pull/4330 is merged.

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/README.md
Comment (Itay4) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/README.md:443]: please update outputs here as well

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/README.md
Comment (Itay4) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/README.md:430]: move the required args to be on top of the optional ones

File: Packs/MicrosoftTeams/Scripts/GrantAccessPolicy/GrantAccessPolicy.yml
Comment (Itay4) [inline Packs/MicrosoftTeams/Scripts/GrantAccessPolicy/GrantAccessPolicy.yml:1]: please add test playbook for this scriipt

File: Packs/MicrosoftTeams/Scripts/GrantAccessPolicy/GrantAccessPolicy.yml
Comment (Itay4) [inline Packs/MicrosoftTeams/Scripts/GrantAccessPolicy/GrantAccessPolicy.yml:32]:
```suggestion
name: ConfigureAzureApplicationAccessPolicy
```

File: Packs/MicrosoftTeams/Scripts/GrantAccessPolicy/GrantAccessPolicy.yml
Comment (Itay4) [inline Packs/MicrosoftTeams/Scripts/GrantAccessPolicy/GrantAccessPolicy.yml:29]:
```suggestion
  id: ConfigureAzureApplicationAccessPolicy
```

File: Packs/MicrosoftTeams/Scripts/GrantAccessPolicy/GrantAccessPolicy.yml
Comment (Itay4) [inline Packs/MicrosoftTeams/Scripts/GrantAccessPolicy/GrantAccessPolicy.yml:36]: what is this tag?

File: Packs/MicrosoftTeams/Scripts/GrantAccessPolicy/GrantAccessPolicy.ps1
Comment (Itay4) [inline Packs/MicrosoftTeams/Scripts/GrantAccessPolicy/GrantAccessPolicy.ps1:1]: @jochman please review the powershell code here

File: Packs/MicrosoftTeams/Scripts/GrantAccessPolicy/GrantAccessPolicy.ps1
Comment (Itay4) [inline Packs/MicrosoftTeams/Scripts/GrantAccessPolicy/GrantAccessPolicy.ps1:20]: need some output here

**Review by jochman — CHANGES_REQUESTED** [review]

File: Packs/MicrosoftTeams/Scripts/GrantAccessPolicy/GrantAccessPolicy.ps1
Comment (jochman) [reply Packs/MicrosoftTeams/Scripts/GrantAccessPolicy/GrantAccessPolicy.ps1]: @gal-berger why you marked it as resolved?

**Review by jochman — CHANGES_REQUESTED** [review]

**Comment by OrelHaim** [comment]
Due to git issue, using other's.
Please review again
the cong pr is in -  https://github.com/demisto/content-test-conf/pull/1167

File: Packs/MicrosoftTeams/Scripts/GrantAccessPolicy/GrantAccessPolicy.yml
Comment (OrelHaim) [reply Packs/MicrosoftTeams/Scripts/GrantAccessPolicy/GrantAccessPolicy.yml]: It's a problem because the user must put it's admin credentials and it's not something we can get from the integration input. besides, we need a special self-service to run it, like the one Tomer created and not just any self-service such as the nightly one.
I'll add to Centrify an example of how to run that script with our credentials.

**Comment by OrelHaim** [comment]
Hi @kirbles19 can you please review it?
We've added a command to the Teams integration and the `ConfigureAzureApplicationAccessPolicy` script.

File: Packs/MicrosoftTeams/Scripts/ConfigureAzureApplicationAccessPolicy/ConfigureAzureApplicationAccessPolicy.ps1
Comment (jochman) [inline Packs/MicrosoftTeams/Scripts/ConfigureAzureApplicationAccessPolicy/ConfigureAzureApplicationAccessPolicy.ps1:25]: If you're outputting the status, you shall enter here the username as well.

Comment (OrelHaim) [reply Packs/MicrosoftTeams/Scripts/ConfigureAzureApplicationAccessPolicy/ConfigureAzureApplicationAccessPolicy.ps1:25]: I was thinking about it but then maybe I should add the app_id too?
There are both inputs to the script, wdut?

Comment (jochman) [reply Packs/MicrosoftTeams/Scripts/ConfigureAzureApplicationAccessPolicy/ConfigureAzureApplicationAccessPolicy.ps1:25]: Well, you need at least one identifier, so just pick on or both.

**Comment by OrelHaim** [comment]
@Itay4 Can you please review?

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/README.md
Comment (Itay4) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/README.md:418]:
```suggestion
`OnlineMeetings.ReadWrite.All`
```

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/README.md
Comment (Itay4) [reply Packs/MicrosoftTeams/Integrations/MicrosoftTeams/README.md:419]: did you review this with @kirbles19 / @ShirleyDenkberg ?

Comment (OrelHaim) [reply Packs/MicrosoftTeams/Integrations/MicrosoftTeams/README.md:419]: not yet, waiting for their approval

**Comment by ShirleyDenkberg** [comment]
@OrelHaim @Itay4 @bakatzir @jochman Doc review completed.

**Review by Itay4 — APPROVED** [review]

**Comment by Itay4** [comment]
build fails on docker image validation due to the fact we dont want to bump it at the moment
merging

**Pull request merged**
